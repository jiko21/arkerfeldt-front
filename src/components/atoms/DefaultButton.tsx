/** @jsxImportSource @emotion/react */
import { css } from '@emotion/react';
import { ButtonHTMLAttributes, FC } from 'react';
import { Color } from '@/const/color';

const buttonStyle = css`
  background-color: ${Color.BLACK};
  border: none;
  border-radius: 6px;
  color: #ffffff;
  font-size: 18px;
  font-weight: 700;
  height: 30px;
  min-width: 240px;
  width: 100%;
  transition: 0.5s;
  &:hover {
    transform: translateY(-4px);
  }
`;

interface IProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  testId?: string;
}

const DefaultButton: FC<IProps> = ({ children, testId, ...props }) => (
  <button css={buttonStyle} {...props} data-testid={testId}>
    {children}
  </button>
);

export default DefaultButton;
