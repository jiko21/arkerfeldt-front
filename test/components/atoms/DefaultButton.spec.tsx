import DefaultButton from '@/components/atoms/DefaultButton';
import { fireEvent, render, screen } from '@testing-library/react';

describe('DefaultButton.tsx', () => {
  let onClickMock: jest.Mock;
  beforeEach(() => {
    onClickMock = jest.fn();
  });

  afterEach(() => {
    onClickMock.mockClear();
  });

  it('should correctly render', () => {
    const { container } = render(
      <DefaultButton onClick={onClickMock}>Sample</DefaultButton>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should correctly call onClick', () => {
    const TEST_ID = 'TEST_ID'
    render(<DefaultButton onClick={onClickMock} testId={TEST_ID}>Sample</DefaultButton>);
    fireEvent.click(screen.getByTestId(TEST_ID));
    expect(onClickMock).toBeCalled();
  });
});
