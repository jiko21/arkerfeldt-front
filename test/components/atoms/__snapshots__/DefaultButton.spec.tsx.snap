// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultButton.tsx should correctly render 1`] = `
.emotion-0 {
  background-color: #2d2d2d;
  border: none;
  border-radius: 6px;
  color: #ffffff;
  font-size: 18px;
  font-weight: 700;
  height: 30px;
  min-width: 240px;
  width: 100%;
  -webkit-transition: 0.5s;
  transition: 0.5s;
}

.emotion-0:hover {
  -webkit-transform: translateY(-4px);
  -moz-transform: translateY(-4px);
  -ms-transform: translateY(-4px);
  transform: translateY(-4px);
}

<div>
  <button
    class="emotion-0"
  >
    Sample
  </button>
</div>
`;
